cmd/influxd/inspect/report_tsm/report_tsm.go

```diff
@@ -0,0 +1,366 @@
+			strconv.FormatInt(int64(dbCount.Count()-oldCount), 10),
+			time.Unix(0, minT).UTC().Format(time.RFC3339Nano),
+			time.Unix(0, maxT).UTC().Format(time.RFC3339Nano),
+			loadTime.String(),
+		}, "\t"))
+		if a.detailed {
+			err = tw.Flush()
+			if err != nil {
+				return fmt.Errorf("failed to flush tabwriter: %v", err)
+			}
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	err := tw.Flush()
+	if err != nil {
+		return fmt.Errorf("failed to flush tabwriter: %v", err)
+	}
+
+	printSummary(cmd, printArgs{
+		fileCount:          fileCount,
+		minTime:            minTime,
+		maxTime:            maxTime,
+		estTitle:           estTitle,
+		totalSeries:        totalSeries,
+		detailed:           a.detailed,
+		tagCardinalities:   tagCardinalities,
+		measCardinalities:  measCardinalities,
+		fieldCardinalities: fieldCardinalities,
+		dbCardinalities:    dbCardinalities,
+	})
+
+	cmd.Printf("Completed in %s\n", time.Since(start))
+	return nil
+}
+
+type printArgs struct {
+	fileCount        int
+	minTime, maxTime int64
+	estTitle         string
+	totalSeries      counter
+	detailed         bool
+
+	tagCardinalities   map[string]counter
+	measCardinalities  map[string]counter
+	fieldCardinalities map[string]counter
+	dbCardinalities    map[string]counter
+}
+
+func printSummary(cmd *cobra.Command, p printArgs) {
+	cmd.Printf("\nSummary:")
+	cmd.Printf("  Files: %d\n", p.fileCount)
+	cmd.Printf("  Time Range: %s - %s\n",
+		time.Unix(0, p.minTime).UTC().Format(time.RFC3339Nano),
+		time.Unix(0, p.maxTime).UTC().Format(time.RFC3339Nano),
+	)
+	cmd.Printf("  Duration: %s \n\n", time.Unix(0, p.maxTime).Sub(time.Unix(0, p.minTime)))
+
+	cmd.Printf("Statistics\n")
+	cmd.Printf("  Series:\n")
+	for db, counts := range p.dbCardinalities {
+		cmd.Printf("     - %s%s: %d (%d%%)\n", db, p.estTitle, counts.Count(), int(float64(counts.Count())/float64(p.totalSeries.Count())*100))
+	}
+	cmd.Printf("  Total%s: %d\n", p.estTitle, p.totalSeries.Count())
+
+	if p.detailed {
+		cmd.Printf("\n  Measurements (est):\n")
+		for _, t := range sortKeys(p.measCardinalities) {
+			cmd.Printf("    - %v: %d (%d%%)\n", t, p.measCardinalities[t].Count(), int((float64(p.measCardinalities[t].Count())/float64(p.totalSeries.Count()))*100))
+		}
+
+		cmd.Printf("\n  Fields (est):\n")
+		for _, t := range sortKeys(p.fieldCardinalities) {
+			cmd.Printf("    - %v: %d\n", t, p.fieldCardinalities[t].Count())
+		}
+
+		cmd.Printf("\n  Tags (est):\n")
+		for _, t := range sortKeys(p.tagCardinalities) {
+			cmd.Printf("    - %v: %d\n", t, p.tagCardinalities[t].Count())
+		}
+	}
+}
+
+// sortKeys is a quick helper to return the sorted set of a map's keys
+func sortKeys(vals map[string]counter) (keys []string) {
+	for k := range vals {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	return keys
+}
+
+func (a *args) walkShardDirs(root string, fn func(db, rp, id, path string) error) error {
+	type location struct {
+		db, rp, id, path string
+	}
+
+	var dirs []location
```

Comment:
To avoid future confusion about potential duplication, I'd suggest renaming this to `tsms`